ynr/apps/parties/importer.py

```diff
@@ -0,0 +1,324 @@
+                "217",
+                "2508",
+                "2510",
+                "2512",
+                "2514",
+                "281",
+                "289",
+                "301",
+                "303",
+                "305",
+                "3560",
+                "37",
+                "38",
+                "4",
+                "404",
+                "410",
+                "445",
+                "49",
+                "60",
+                "62",
+                "68",
+                "74",
+            ],
+        }
+
+    def get_party_list(self, start):
+        params = self.params
+        params["start"] = start
+        url = "{}?{}".format(self.base_url, urlencode(params, doseq=True))
+        req = requests.get(url)
+        req.raise_for_status()
+        return req.json()
+
+    def do_import(self):
+        start = 0
+        party_list = self.get_party_list(start)
+        while start <= int(party_list["Total"]):
+            for party_dict in party_list["Result"]:
+                party = ECParty(party_dict)
+                party_model, created = party.save()
+                if created:
+                    self.collector.append(party_model)
+            start = start + self.per_page
+            party_list = self.get_party_list(start)
+        return self.collector
+
+    def create_joint_parties(self):
+        """
+        We create joint parties if one party's description matches the
+        JOINT_DESCRIPTION_REGEX (e.g. contains "joint description").
+
+        The new party gets an ID made up of both parties IDs like
+
+        `joint-party:51-83`
+
+        Where each number is the ID of the 'parent' party.
+
+        The lower number ID is always first.
+
+        """
+
+        qs = PartyDescription.objects.filter(
+            description__iregex=JOINT_DESCRIPTION_REGEX
+        )
+
+        for description in qs:
+            joint_party_name, other_party_names = re.match(
+                JOINT_DESCRIPTION_REGEX, description.description, re.IGNORECASE
+            ).groups()
+
+            sub_parties = other_party_names.split(", and ")
+            for other_party_name in sub_parties:
+                # Try to get a corrected name for the other party
+                other_party_name = other_party_name.replace("’", "'")
+                other_party_name = CORRECTED_PARTY_NAMES_IN_DESC.get(
+                    other_party_name, other_party_name
+                )
+                date = CORRECTED_DESCRIPTION_DATES.get(
+                    description.description,
+                    description.date_description_approved,
+                )
+
+                try:
+                    other_party = (
+                        Party.objects.filter(
+                            name=other_party_name,
+                            register=description.party.register,
+                        )
+                        .active_for_date(date)
+                        .get()
```

Comment:
Is chaining an empty `.get()` on the end here a shorthand to `raise` if we have more than one record?

Reply:
Cool - I've just not seen that syntax before. I'll have to remember it